### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by ad9311**
### Greetings! 👋🏽 

In this PR, I present the following changes:

- When the user clicks a task checkbox, the description gets strikethrough and, the object's key for completed is set to true.
- If the user unchecks a checkbox, the previously mentioned changes are reverted.
- The array is stored in the local storage.
- If the user refreshes the page since the changes are stored in local storage they are still displayed.
- Split functions into more files.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
-  renderList();
-});
+function updateTask(e) {
+  if (e.target !== e.currentTarget) {
+    const index = e.target.id.replace(/^\D+/g, '');
+    const completed = e.target.checked;
+    list.setCompleted(index, completed);
+    element.updateDescription(`desc${index}`, completed);
+    updateLocalStorage();
+  }
```
Comment (hemant-soni-vst-au4) [inline src/index.js:33]: - [ ] As per requirements, The task object's values for completion are updated upon user actions. The updated array of items is stored in local storage, so the user gets the correct list values after the page reloads. But after making changes for task completion when I refresh the page it gets me back to the initial task position. Kindly fix this issue by storing the updated task value in localStorage.

File: src/index.js [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+    ul.appendChild(element.createListTask(list.list[i]));
   }
+  updateLocalStorage();
 }
 
-window.addEventListener('load', () => {
-  renderList();
-});
+function updateTask(e) {
+  if (e.target !== e.currentTarget) {
```
Comment (hemant-soni-vst-au4) [inline src/index.js:27]: - [ ] As per the learning objective we have to follow the guidelines of ES6 syntax. Kindly try to use the arrow function.

File: src/index.js [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
-];
+list.addNewTask('Walk dog');
+list.addNewTask('Clean kitchen');
+list.addNewTask('Pay bills');
+list.addNewTask('Water plants');
+list.addNewTask('Buy food');
+
+function updateLocalStorage() {
+  localStorage.setItem('list', JSON.stringify(list.list));
+}
```
Comment (hemant-soni-vst-au4) [inline src/index.js:13]: - [ ] As per the learning objective we have to follow the guidelines of ES6 syntax. Kindly try to use the arrow function.

**Review by hemant-soni-vst-au4 — CHANGES_REQUESTED** [review]
Hi @ad9311 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there! 

Alt Text
## Required Changes ♻️
_Check the comments under the review._

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ad9311** [comment]
As per requested by the preivous reviewer, I made the following changes:
- Changed functions in index.js to arrow functions to follow the guidelines of ES6 syntax.
- If the user refreshes the page since the changes are stored in local storage they are still displayed. (This was implemented before the first review but after submiting for a review).

**Review by hemant-soni-vst-au4 — APPROVED** [review]
Hi @ad9311, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**